pkg/edgeview/src/basics.go

```diff
@@ -0,0 +1,665 @@
+			uuidStr = strings.TrimSuffix(string(retbytes), "\n")
+		}
+	}
+
+	if jdata.Exp == 0 || jdata.Dep == "" {
+		return addrport, path, fmt.Errorf("read JWT data failed")
+	}
+	if uuidStr != "" && jdata.Sub != uuidStr {
+		return addrport, path, fmt.Errorf("uuid does not match JWT jti")
+	}
+
+	now := time.Now()
+	nowSec := uint64(now.Unix())
+	if nowSec > jdata.Exp {
+		return addrport, path, fmt.Errorf("JWT expired %d sec ago", nowSec - jdata.Exp)
+	}
+
+	if jdata.Num > 1 && instID < 1 {
+		return addrport, path, fmt.Errorf("Edgeview is in multi-instance mode, inst-id(1-%d) needs to be specified", jdata.Num)
+	} else if jdata.Num == 1 && instID > 0 {
+		return addrport, path, fmt.Errorf("Edgeview is not in multi-instance mode, no need to specify inst-ID")
+	}
+
+	if strings.Contains(jdata.Dep, "/") {
+		urls := strings.SplitN(jdata.Dep, "/", 2)
+		addrport = urls[0]
+		path = "/" + urls[1]
+	} else {
+		addrport = jdata.Dep
+	}
+
+	evStatus.ExpireOn = jdata.Exp
+	evStatus.StartedOn = now
+	encryptVarInit(jdata)
+
+	return addrport, path, nil
+}
+
+func checkClientIPMsg(msg string) bool {
+	if strings.HasPrefix(msg, clientIPMsg) {
+		msgs := strings.SplitN(msg, clientIPMsg, 2)
+		if len(msgs) == 2 {
+			addrPort := strings.Split(msgs[1], ":")
+			if net.ParseIP(addrPort[0]) == nil {
+				log.Errorf("received invalid IP %v", msg)
+				return false
+			}
+			if len(addrPort) == 2 {
+				num, err := strconv.Atoi(addrPort[1])
+				if err != nil || num < 1024 || num > 65535 {
+					log.Errorf("received invalid IP %v", msg)
+					return false
+				}
+			} else if len(addrPort) > 2 {
+				log.Errorf("received invalid IP %v", msg)
+				return false
+			}
+			myEvEndPoint = msgs[1]
+			return true
+		}
+	}
+	return false
+}
+
+func evLogger(deb bool) *logrus.Logger {
+	formatter := logrus.JSONFormatter{
+		TimestampFormat: time.RFC3339Nano,
+	}
+	logger := logrus.New()
+	logger.SetFormatter(&formatter)
+	log = base.NewSourceLogObject(logger, agentName, os.Getpid())
+	if deb {
+		logger.SetLevel(logrus.DebugLevel)
+	}
+	return logger
+}
+
+func getBasics() {
+	if runOnServer {
+		if _, err := os.Stat("/config"); err != nil {
+			return
+		}
+	}
+
+	ips := []string{}
+	for _, i := range allUPIntfIPv4() {
+		if strings.HasPrefix(i.intfName, "bn") {
+			continue
+		}
+		ips = append(ips, i.ipAddr)
+	}
+	if myEvEndPoint != "" {
+		fmt.Printf("Device IPs: %v; Endpoint IP %s\n", ips, myEvEndPoint)
+	} else {
+		fmt.Printf("Device IPs: %v\n", ips)
+	}
+
+	retbytes, err := ioutil.ReadFile("/persist/status/uuid")
+	if err == nil {
+		fmt.Printf("  UUID: %s", string(retbytes))
+	}
+	retbytes, err = ioutil.ReadFile("/config/server")
+	if err == nil {
+		var printed bool
+		serverurl := string(retbytes)
+		conts := strings.Split(serverurl, "zedcloud.")
+		if len(conts) == 2 {
+			cont2s := strings.Split(conts[1], ".zededa.net")
+			if len(cont2s) == 2 {
+				cluster := cont2s[0]
+				controller := strings.Replace(serverurl, cluster, "\033[0;93m"+cluster+"\033[0m", 1)
```

Comment:
What does this do adding the control characters? Color??

Reply:
yes, a non production cluster name is highlighted in color.

Reply:
Wouldn't be a bad idea to add comments about the color codes. I see you have some color function but there is also code which adds control characters here and there.

Reply:
added a new func 'getColorStr()' to handle string replacement.